pyo3-macros-backend/src/pyclass.rs

```diff
@@ -691,19 +706,20 @@ fn impl_simple_enum(
 
     let (default_repr, default_repr_slot) = {
         let variants_repr = variants.iter().map(|variant| {
+            ensure_spanned!(variant.options.constructor.is_none(), variant.options.constructor.span() => "`constructor` can't be used on a simple enum variant");
```

Comment:
It might be worth moving this check into its own loop just above rather than merging it with the generation of `__repr__`, both for clarity and to avoid breaking the check should we change the `__repr__` generation later.